tests/saw/spec/handshake/rfc_handshake_tls13.cry

```diff
@@ -0,0 +1,266 @@
+            (map rfc2S2N (doHandshake`{len} params))
+            (traceS2N`{len} conn)
+
+type MessageType = [4]
+(applicationData        : MessageType) = 0
+(clientHello            : MessageType) = 1
+(helloRetryRequest      : MessageType) = 2
+(serverHello            : MessageType) = 3
+(encryptedExtensions    : MessageType) = 4
+(certificateRequest     : MessageType) = 5
+(certificate            : MessageType) = 6
+(certificateVerify      : MessageType) = 7
+(finished               : MessageType) = 8
+(endOfEarlyData         : MessageType) = 9
+(changeCipherSpec       : MessageType) = 10
+(error                  : MessageType) = 12
+
+type Sender = [2]
+(server   : Sender) = 0
+(client   : Sender) = 1
+(both     : Sender) = 2
+
+type Message = {
+    messageType : MessageType,
+    sender : Sender
+}
+
+type Parameters = {
+    compatMode : [2],
+    psk_mode : Bit,
+    retry : Bit,
+    client_auth : Bit,
+    no_client_cert : Bit,
+    zero_rtt : Bit
+}
+
+connectionParameters : connection -> Parameters -> Bit
+connectionParameters conn params = params.psk_mode == conn_psk_mode
+                                   /\ params.retry == conn_retry
+                                   /\ params.client_auth == conn_client_auth /\ ~params.psk_mode
+                                   /\ params.no_client_cert == conn_no_client_cert /\ params.client_auth
+                                   /\ params.zero_rtt == conn_zero_rtt /\ params.psk_mode
+                                   /\ params.compatMode == [ conn.mode == S2N_SERVER, conn.mode == S2N_CLIENT ]
+                                   /\ conn.actual_protocol_version == S2N_TLS13
+  where conn_psk_mode =         False /* Not supported yet */
+        conn_retry =            False /* Not supported yet */
+        conn_client_auth =      False /* Not supported yet */
+        conn_no_client_cert =   False /* Not supported yet */
+        conn_zero_rtt =         False /* Not supported yet */
+
+type Action = {
+    message : Message,
+    valid : Bit
+}
+
+type StateId = [5]
+type State = {
+    actions : [5]Action,
+    id : StateId,
+    next : StateId
+}
+
+doHandshake : {n} (fin n, n >= 1) => Parameters -> [n]Message
+doHandshake params = (foldl doState initialHandshake (take`{n} (stateMachine params))).messages
+  where doState hs state = foldl doAction hs state.actions
+        doAction hs action = if hs.index < `n /\ action.valid
+                             then updateHandshake hs action.message
+                             else hs
+        updateHandshake hs value = {
+            messages = update`{ix=(width n)} hs.messages hs.index value,
+            index = hs.index + 1
+        }
+        initialHandshake = {
+            messages = repeat { messageType = applicationData, sender = both },
+            index = zero
+        }
+
+/*
+ * State machine based on RFC 8446, specifically:
+ * - The basic state machines in Appendix A
+ * - The middlebox compatability rules in Appendix D.4
```

Comment:
The proof covers the messages that are required to be *sent* during the handshake under various conditions. It does not prove that the client or server correctly ignores cipher_change_spec messages that may be *received* at any time during the handshake per [RFC8446 D.4](https://tools.ietf.org/html/rfc8446#appendix-D.4). From offline discussion w/ lrstewart: the correct behavior of the s2n implementation is covered by [unit tests for client and server](https://github.com/awslabs/s2n/blob/master/tests/unit/s2n_tls13_state_machine_handshake_test.c#L234-L306). Yay!